aten/src/ATen/native/sparse/SparseTensor.cpp

```diff
@@ -518,32 +518,35 @@ const SparseTensor& resize_as_sparse_(const SparseTensor& self, const SparseTens
   return self;
 }
 
-SparseTensor dense_to_sparse(const Tensor& self, c10::optional<c10::Layout> layout, OptionalIntArrayRef blocksize) {
+SparseTensor dense_to_sparse(const Tensor& self, c10::optional<c10::Layout> layout, OptionalIntArrayRef blocksize, c10::optional<int64_t> dense_dim_opt) {
   if (layout.has_value()) {
     if (blocksize.has_value() && !(*layout == kSparseBsr || *layout == kSparseBsc)) {
       AT_ERROR("to_sparse for ", self.layout(), " to ", *layout, " conversion does not use specified blocksize");
     }
     if (self.layout() == *layout) {
       return self;
     }
+    if (dense_dim_opt.has_value() && !(*layout == kSparseCsr || *layout == kSparseCsc || *layout == kSparseBsr || *layout == kSparseBsc)) {
+      AT_ERROR("to_sparse for ", self.layout(), " to ", *layout, " conversion does not support specifying number of dense dimensions");
+    }
```

Comment:
To support COO layout and `dense_dim` combination:
```suggestion
```
because `dense.to_sparse(layout=torch.strided, dense_dim=dense_dim) -> dense` for any `dense_dim` value by definition.

Reply:
Changed according to all of the suggestions.